lib/imperative-test.js

```diff
@@ -0,0 +1,396 @@
+'use strict';
+
+const compare = require('./compare');
+const Test = require('./test');
+const { instance: runnerInstance } = require('./runner');
+
+const defaultTestOptions = {
+  run: true,
+  async: true,
+  timeout: 30000,
+  context: {},
+  parallelSubtests: false,
+};
+
+class ImperativeTest extends Test {
+
+  // TODO(lundibundi): Current implementaion will not fail if
+  // you have any tests past the 'plan' or after 'end' but it should
+
+  constructor(caption, func, options) {
+    super(caption, options);
+
+    Object.keys(defaultTestOptions).filter(k => this.options[k] === undefined)
```

Comment:
Maybe move `filter` to a next line, same as `DeclarativeTest`(https://github.com/metarhia/metatests/pull/66/files?w=1#diff-8211b5f849d240615ffa958a3ae13024R21)?